Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,232 @@
+/-
+Copyright (c) 2023 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Analysis.Convex.Combination
+import Mathlib.LinearAlgebra.AffineSpace.Independent
+import Mathlib.Tactic.Linarith
+
+/-!
+# Radon's convexity theorem
+
+Radon's theorem on convex sets states that any affine dependent set can be partitioned into two sets
+whose convex hulls intersect.
+
+## Main results
+
+* `Radon_partition`: Radon convexity theorem
+
+## Tags
+convex hull, radon
+
+-/
+
+
+open Set Finset
+open BigOperators
+
+universe u
+
+variable {𝕜 : Type*} {E : Type u} [LinearOrderedField 𝕜] [AddCommGroup E] [Module 𝕜 E]
+
+/-- Any family `f` of affine dependent vectors contains a set `I` with the property that
+convex hulls of `I` and `Iᶜ` intersect. -/
+theorem radon_partition {ι : Type*} {f : ι → E}
+    (h : ¬AffineIndependent 𝕜 f) : ∃ (I : Set ι),
+    (Set.Nonempty ((convexHull 𝕜 (f '' I)) ∩ (convexHull 𝕜 (f '' Iᶜ)))) := by
+  unfold AffineIndependent at h; push_neg at h
+  rcases h with ⟨s1, w, h_wsum, h_vsum, nonzero_w_index, h1, h2⟩
+  let I : Finset ι := s1.filter (fun i => w i ≥ 0)
```

Comment:
```suggestion
  let I : Finset ι := s1.filter (fun i => 0 ≤ w i)
```
This may sound very picky, but there is a good reason: by convention, most lemmas in the library use `≤` instead of `≥`. The choice is a bit arbitrary, but it's important that we have a "normal form", otherwise we would spend a lot of time reversing the inequality back-and-forth depending on the lemmas we use. So it's best to stick to it.